### PR TITLE
Device factory clone device

### PR DESCRIPTION
**Pull request by marcin-usielski**

### CODE REVIEW

File: test/device/test_device_configuration.py [hunk 1]
```diff
@@ -142,6 +142,23 @@ def test_can_select_device_loaded_from_config_file(moler_config, device_factory)
     assert device.__class__.__name__ == 'UnixLocal'
 
 
+def test_can_clone_device(moler_config, device_factory):
+    conn_config = os.path.join(os.path.dirname(__file__), os.pardir, "resources", "device_config.yml")
+    moler_config.load_config(config=conn_config, config_type='yaml')
+
+    device_org = device_factory.get_device(name='UNIX_LOCAL')
+    assert device_org is not None
+    device_cloned = device_factory.get_cloned_device(source_device=device_org, new_name="CLONED_UNIX_LOCAL")
```
Comment (greg-latuszek) [inline test/device/test_device_configuration.py]: Why we need `source_device` as parameter? Notice that `device_factory.get_device()` besides returning device is caching it. Wouldn't it be simpler just to call:
```python
device_cloned = device_factory.get_cloned_device(source_device_name="UNIX_LOCAL", new_name="CLONED_UNIX_LOCAL")
```

Comment (marcin-usielski) [reply test/device/test_device_configuration.py]: Good point.

File: moler/device/device.py [hunk 2]
```diff
@@ -49,6 +50,23 @@ def get_device(cls, name=None, device_class=None, connection_desc=None, connecti
                                      establish_connection=establish_connection)
         return dev
 
+    @classmethod
+    def get_cloned_device(cls, source_device, new_name, initial_state=None, establish_connection=True):
+        if new_name in cls._devices.keys():
+            return cls._devices[new_name]  # or raise exception?
```
[the remark this answers is not in the document]
Comment (marcin-usielski) [reply moler/device/device.py]: Added checking source device when attempt to clone device with the same name as previous cloned device.

File: test/resources/device_config.yml [hunk 3]
```diff
@@ -21,6 +21,9 @@ DEVICES:
     DEVICE_CLASS: moler.device.unixlocal.UnixLocal
     INITIAL_STATE: UNIX_LOCAL
 
+  UNIX_LOCAL_CLONED_VIA_YAML:
+    CLONED: UNIX_LOCAL
```
Comment (greg-latuszek) [inline test/resources/device_config.yml]: I would change name to `CLONE_OF` or `CLONED_FROM`. Current naming is little bit confusing. You may read it as: **UNIX_LOCAL_CLONED_VIA_YAML has cloned UNIX_LOCAL (and what? stored it somewhere? thrown it away?)**

However, sentences
* UNIX_LOCAL_CLONED_VIA_YAML is `clone of` UNIX_LOCAL or 
* UNIX_LOCAL_CLONED_VIA_YAML has been `cloned from` UNIX_LOCAL
are self explanatory.

Comment (marcin-usielski) [reply test/resources/device_config.yml]: Changed into **CLONED_FROM**

**Pull request merged**